assets/pages/new-contributions-landing/components/StripePaymentRequestButton.jsx

```diff
@@ -0,0 +1,198 @@
+  countryGroupId: CountryGroupId,
+  amount: number,
+|};
+
+
+const mapStateToProps = (state: State) => ({
+  selectedAmounts: state.page.form.selectedAmounts,
+  otherAmounts: state.page.form.formData.otherAmounts,
+  canMakeApplePayPayment: state.page.form.stripePaymentRequestButtonData.canMakeApplePayPayment,
+  paymentRequest: state.page.form.stripePaymentRequestButtonData.paymentRequest,
+  countryGroupId: state.common.internationalisation.countryGroupId,
+  country: state.common.internationalisation.countryId,
+  currency: state.common.internationalisation.currencyId.toLowerCase(),
+  isTestUser: state.page.user.isTestUser || false,
+  contributionType: state.page.form.contributionType,
+});
+
+const mapDispatchToProps = (dispatch: Function) => ({
+  onPaymentAuthorised:
+    (token, complete) => { dispatch(onStripePaymentRequestApiPaymentAuthorised(token, complete)); },
+  setCanMakeApplePayPayment:
+    (canMakeApplePayPayment) => { dispatch(setCanMakeApplePayPayment(canMakeApplePayPayment)); },
+  setPaymentRequest:
+    (paymentRequest) => { dispatch(setPaymentRequest(paymentRequest)); },
+  updateEmail: (email: string) => { dispatch(updateEmail(email)); },
+  setStripePaymentRequestButtonClicked: () => { dispatch(setStripePaymentRequestButtonClicked()); },
+});
+
+
+// ----- Functions -----//
+
+
+function updateUserEmail(data: Object, setEmail: string => void) {
+  const email = data.payerEmail;
+  if (email) {
+    if (isValidEmail(email)) {
+      setEmail(email);
+    } else {
+      logException(`Failed to set email for stripe payment request user with email: ${email}`);
+    }
+  } else {
+    logException('Failed to set email: no email in data object');
+  }
+}
+
+const onComplete = (complete: Function) => (res: PaymentResult) => {
+  if (res.paymentStatus === 'success') {
+    complete('success');
+  } else if (res.paymentStatus === 'failure') {
+    complete('fail');
+  }
+};
+
+function onClick(event, props: PropTypes) {
+  event.preventDefault();
+  props.setStripePaymentRequestButtonClicked();
+  const amountIsValid =
+    checkAmountOrOtherAmount(
+      props.selectedAmounts,
+      props.otherAmounts,
+      props.contributionType,
+      props.countryGroupId,
+    );
+  if (props.paymentRequest && amountIsValid) {
+    props.paymentRequest.show();
+  }
+}
+
+function initialisePaymentRequest(props: PropTypes) {
+  const paymentRequest = props.stripe.paymentRequest({
+    country: props.country,
+    currency: props.currency,
+    total: {
+      label: 'The Guardian',
+      amount: props.amount,
+    },
+    requestPayerEmail: true,
+  });
+  paymentRequest.on('token', ({ complete, token, ...data }) => {
+    // We need to do this so that we can offer marketing permissions on the thank you page
+    updateUserEmail(data, props.updateEmail);
+    const tokenId = props.isTestUser ? 'tok_visa' : token.id;
+    props.onPaymentAuthorised({ paymentMethod: 'Stripe', token: tokenId }, onComplete(complete));
```

Comment:
Not necessary for this PR, but it would be good to have a re-think and see if we can return a promise from async actions like `onStripePaymentRequestApiPaymentAuthorised `. Then we would need to pass in this `onComplete` callback.

It certainly seems like it's possible, looking at e.g. https://redux.js.org/advanced/async-actions#indexjs

But not sure off the top of my head how we do something similar (might not require major changes)